### DISCUSSION TITLE
Realtime works for INSERT but not DELETE

### DISCUSSION THREAD

**Discussion by cdedreuille** [supabase/supabase#1521] [Questions]
Hi,

I'm having a weird bug where the realtime listener works for INSERT but not DELETE. Here is my code:
```
  useEffect(() => {
    const mySubscription = supabase
      .from(`occasionProducts:occasion=eq.${occasion.id}`)
      .on('*', (payload) => {
        console.log('Change received!', payload);
      })
      .subscribe();

    return () => supabase.removeSubscription(mySubscription);
  }, []);
```
If I'm listening to the entire table both INSERT and DELETE works but using `occasionProducts:occasion=eq.${occasion.id}`, only INSERT works but not DELETE.

Any idea what I'm doing wrong here?

**Comment by inian** [comment]
Try setting the replica identity to full as mentioned [here](https://supabase.io/docs/guides/database#realtime) @cdedreuille 
```
alter table_name your_table 
replica identity full;
```

**Reply by cdedreuille** [reply]
Thanks for your answer. I'm sorry very noob question as I'm quite new to SQL:
- What values should I put instead of `table_name` and `your_table`?
Should it be `alter occasionProducts occasionProducts` ?
- I'm not sure I understand what this will actually do as if I listen to the full table, I actually receive something when I delete a row.
- If I go to my dashboard, it seems everything is already ticked for all tables, so why would it not work if I'm listening to specific rows?

<img width="1117" alt="Screenshot 2021-05-11 at 09 09 42" src="https://user-images.githubusercontent.com/1540635/117781603-bafd3080-b238-11eb-9940-873f03ba47b6.png">

**Reply by soedirgo** [reply]
Hey Charles,
- That seems like a typo in the docs, it should be `ALTER TABLE "occasionProducts" REPLICA IDENTITY FULL;`
- With inian's suggestion above, if it works for the full table it should work for specific rows
- The settings on the dashboard looks fine to me

My guess is the `occasionProducts` table has `REPLICA IDENTITY DEFAULT` (only records primary key from old values) and `occasion` isn't the primary key. So the filter with `eq` doesn't apply.
